src/operations/operation.ts

```diff
@@ -34,23 +34,42 @@ export interface OperationOptions extends BSONSerializeOptions {
  * a specific aspect.
  * @internal
  */
-export abstract class OperationBase<
-  T extends OperationOptions = OperationOptions,
-  TResult = Document
-> {
-  options: T;
+export abstract class OperationBase<TResult = Document> {
   ns!: MongoDBNamespace;
   cmd!: Document;
-  readPreference: ReadPreference;
+  readPreference!: ReadPreference;
   server!: Server;
+  session?: ClientSession;
   fullResponse?: boolean;
+  /** Explain returns the aggregation execution plan (requires mongodb 2.6 \>) */
+  explain?: boolean;
+
+  promoteBuffers?: boolean;
+  promoteValues?: boolean;
+  promoteLongs?: boolean;
+  raw?: boolean;
+  ignoreUndefined?: boolean;
 
   // BSON serialization options
-  bsonOptions?: BSONSerializeOptions;
+  get bsonOptions(): BSONSerializeOptions {
+    const bsonOptions: Document = {
+      promoteBuffers: this.promoteBuffers,
+      promoteValues: this.promoteValues,
+      promoteLongs: this.promoteLongs,
+      raw: this.raw,
+      ignoreUndefined: this.ignoreUndefined
+    } as const;
+
+    for (const key in bsonOptions) {
+      bsonOptions[key] === undefined && delete bsonOptions[key];
```

Comment:
Can we do this without using `delete`? I believe we try to stay away from it when possible.

Reply:
I do like this getter BTW 🤩